### PR TITLE
TM-5412, Edit Assignment Cycles

### PR DESCRIPTION
**Pull request by M00staff**
Dual Merge: 
- [API PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/941)
- [Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/444)

[Ticket](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/444)

[Designs](https://6bse6q.axshare.com/#id=jjsxp9&p=cycle_search_-_position_management&g=1)
[Designs](https://6bse6q.axshare.com/#id=c4aoi7&p=assignment_cycle_details&g=1)
[Spreadsheet100](https://usdos-my.sharepoint.com/:x:/r/personal/phungvl_state_gov/_layouts/15/doc2.aspx?sourcedoc=%7B864df564-41fb-43df-9f13-0f8e6bf9b297%7D&action=edit&activeCell=%27Assignment%20Cycles%27!F32&wdinitialsession=4a11d356-0759-4c72-a73e-1a9bc091b609&wdrldsc=15&wdrldc=1&wdrldr=AccessTokenExpiredWarning%2CRefreshingExpiredAccessT)

Link to Cycle Details
image

image

image

There's a lot left to be desired, comments in the PR...

### CODE REVIEW

File: src/Components/AoPage/AoPage.jsx [hunk 1]
```diff
@@ -23,6 +24,7 @@ const AoPage = () => (
       <Route path="/profile/ao/availablebidders" render={() => <AvailableBidderContainer isCDO={false} isAO />} />
       <Route path="/profile/ao/panelmeetingagendas/:pmID" render={() => <PanelMeetingAgendas isAO />} />
       <Route path="/profile/ao/cyclemanagement" render={() => <CycleManagement isAO />} />
+      <Route path="/profile/ao/assginmentcycle/:id" render={() => <AssignmentCycleEdit />} />
```
Comment (M00staff) [inline src/Components/AoPage/AoPage.jsx:27]: i don't think there is a difference between what an AO & Bureau see on this page - so no AO prop gets passed in

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 2]
```diff
@@ -0,0 +1,637 @@
+/* eslint-disable max-len */
```
Comment (M00staff) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:1]: i despise this rule

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 3]
```diff
@@ -0,0 +1,637 @@
...
+    dispatch(cycleManagementAssignmentCycleFetchData(match?.params?.id));
+  }, []);
+
+  useEffect(() => {
+    if (assignmentCycleUpdateSuccess) {
+      setEditMode(false);
+      dispatch(cycleManagementUpdateCycleSuccess(false));
+      dispatch(cycleManagementAssignmentCycleFetchData(match?.params?.id));
+    }
+  }, [assignmentCycleUpdateSuccess]);
```
Comment (M00staff) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:74]: this whole pattern going back to the action/reducer is just to get out of edit mode & make the page reload when an update to the cycle is successful or if the Post Open Positions call is successful.

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 4]
```diff
@@ -0,0 +1,637 @@
...
+    setBidHtf(assignmentCycle?.dates_mapping?.BIDHTF?.begin_date);
+    setBidMds(assignmentCycle?.dates_mapping?.BIDMDS?.begin_date);
+    setBidOrg(assignmentCycle?.dates_mapping?.BIDORG?.begin_date);
+    setBidStart(assignmentCycle?.dates_mapping?.BIDSTART?.begin_date);
+    setBurBid(assignmentCycle?.dates_mapping?.BURBID?.begin_date);
+    setBurEarly(assignmentCycle?.dates_mapping?.BUREARLY?.begin_date);
+    setBurPos(assignmentCycle?.dates_mapping?.BURPOS?.begin_date);
+    setBurPrebd(assignmentCycle?.dates_mapping?.BURPREBD?.begin_date);
+    setAssignedBidderDate(assignmentCycle?.dates_mapping?.PANLASG?.begin_date);
+  }, [assignmentCycleLoading]); // but why?
```
Comment (M00staff) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:99]: this was driving me nuts!

When the Tabbed Component renders, the `readView` & `editView` variables won't have updated data & state -- so the page kept rendering with no data, even though the API call had returned. 

So, all the default states I'm setting between lines 35-60 ^^ above, aren't really doing anything. 

Workaround was this - to update the state once the loading finished - i figured it was some kind of race condition, or i could be blanking on something super simple i overlooked.  Def needs to be looked at & tinkered with 🤔

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 5]
```diff
@@ -0,0 +1,637 @@
...
+      organizationCountReview: bidOrg,
+      biddingStart: bidStart,
+      bureauBidReview: burBid,
+      bureauEarlySeasonBidReview: burEarly,
+      bureauPositionReview: burPos,
+      bureauPreSeasonBidReview: burPrebd,
+      assignedBidder: assignedBidderDate,
+    };
+    dispatch(cycleManagementUpdateCycle(cycleData));
+  };
```
Comment (M00staff) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:129]: my naming of just about everything leaves a lot to be desired - I'm attempting to re-use the `create_assignment_cycle_req_mapping` code in the API for the update call, so mapping here to make sure it fits. 

Still need to test this in dev to see if it works - but i couldn't get the endpoint working in swagger so 🤷

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 6]
```diff
@@ -0,0 +1,637 @@
...
+                />
+              </span>
+            </div>
+          </>
+        }
+      </form>
+      <div>
+        <button onClick={saveAC}>Save and Return</button>
+        {/* <button onClick={deleteAC}>Delete Assignment Cycle</button> */}
+        <button onClick={postAC} type="submit">Post Open Positions</button>
```
Comment (M00staff) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:581]: the Post Open Positions button - just posts the positions for the Assignment Cycle... As of now, the only rule i noticed was that "The Cycle must be Open," for it to work ... Though, 'open' isn't a category or a status 🤔

I think it's referring to the status, though
image

Still brings up the question of either some kind of validation needed - or error handling to let the user know why its failing .. open to adding either

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 7]
```diff
@@ -0,0 +1,637 @@
...
+                  onChange={(date) => setAssignedBidderDate(date)}
+                />
+              </span>
+            </div>
+          </>
+        }
+      </form>
+      <div>
+        <button onClick={saveAC}>Save and Return</button>
+        {/* <button onClick={deleteAC}>Delete Assignment Cycle</button> */}
```
Comment (M00staff) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:580]: There is a delete functionality on the spreadsheet, but not on the designs - gonna follow up with Tim about it ... can add if need be.

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 8]
```diff
@@ -0,0 +1,637 @@
...
+        </div>
+      </div>
+      {
+        getOverlay() ||
+        <div className="ace-wrapper">
+          <TabbedCard
+            tabs={[{
+              text: 'Cycle Information',
+              value: 'INFORMATION',
+              content: editMode ? editView() : readView,
```
Comment (M00staff) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:615]: i couldn't decide which pattern to go with and couldn't remember the difference off the top of my head, so i just went with both and moved on as quickly as possible 😂

originally thought this was the issue with the state not loading, but it wasn't

File: src/Components/CycleManagement/CycleSearchCard.jsx [hunk 9]
```diff
@@ -1,24 +1,10 @@
...
 import { Column, Row } from 'Components/Layout';
 import { formatDate, userHasPermissions } from 'utilities';
-import {
-  deleteAssignmentCyclesSelections,
-  postAssignmentCyclesSelections,
-  saveAssignmentCyclesSelections,
-} from 'actions/cycleManagement';
-import CheckBox from '../CheckBox';
 
 const CycleSearchCard = (props) => {
```
Comment (M00staff) [inline src/Components/CycleManagement/CycleSearchCard.jsx:7]: The original Ui had the Cycle Cards expanding to show the dates, extra info & edit function - the designs had them on a separate page ... I opted to put them on a separate page, cause in order to get the dates & other info, we'd have to make a second API call 

.. this was just much simpler and neater, plus, it was setup so you could have multiple cards open at the same time & multiple card in edit mode, and (again) just opted out of that for simplicity

File: src/actions/cycleManagement.js [hunk 10]
```diff
@@ -287,6 +248,60 @@ export function cycleManagementAssignmentCycleFetchData() {
 }
 
 
+// ================  Cycle Management UPDATE cycle  ================
+
+export function cycleManagementUpdateCycleSuccess(bool) {
+  return {
+    type: 'CYCLE_MANAGEMENT_ASSIGNMENT_CYCLE_UPDATE_SUCCESS',
+    success: bool,
+  };
```
Comment (M00staff) [inline src/actions/cycleManagement.js:257]: goes back to the useEffect to get out of editMode & reload the page

File: src/actions/cycleManagement.js [hunk 11]
```diff
@@ -527,74 +500,7 @@ export function assignmentCycleDeleteDataSuccess(results) {
...
-            dispatch(assignmentCyclePostDataErrored(true));
-            dispatch(assignmentCyclePostDataLoading(false));
-          });
-        }
-      });
-  };
-}
-
-export function deleteAssignmentCyclesSelections(id) {
+export function deleteAssignmentCyclesSelections(id) { // rename.
```
Comment (M00staff) [inline src/actions/cycleManagement.js:503]: will circle back to see if the delete is needed or not

File: src/actions/cycleManagement.js [hunk 12]
```diff
@@ -527,74 +500,7 @@ export function assignmentCycleDeleteDataSuccess(results) {
   };
 }
 
-
-export function assignmentCycleFetchData(query = {}) {
```
Comment (M00staff) [inline src/actions/cycleManagement.js:531]: there were so many actions in this file that it was hard to parse what was for what - especially after starting to gut a lot of it & move things around... ended up just writing new ones, using longer and (hopefully) more clear naming

File: src/Components/CycleManagement/CycleSearchCard.jsx [hunk 13]
```diff
@@ -28,186 +14,23 @@ const CycleSearchCard = (props) => {
...
   return (
     <Row fluid className="cycle-search-card box-shadow-standard">
       <Row fluid className="cyc-card--row">
         <Column columns={3}>
-          {name}
+          {isSuperUser // only Admin?
+            ? <Link to={cycleLink}>
+              <div>{ name }</div>
+            </Link>
+            : <div>{ name }</div>
```
Comment (M00staff) [inline src/Components/CycleManagement/CycleSearchCard.jsx:32]: So, I believe that only the Admin can `edit` a cycle .. but what about just getting to the page with more info (will double check with Tim)

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 14]
```diff
@@ -0,0 +1,637 @@
...
+                e.preventDefault();
+                setEditMode(!editMode);
+              }}
+            >
+              <div>
+                <FA className="fa-solid fa-pencil" />
+                  Edit
+              </div>
+            </Link>
+          }
```
Comment (M00staff) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:201]: only Admin can Edit

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 15]
```diff
@@ -0,0 +1,637 @@
...
+        <div className="ac-sections">
+          <label htmlFor="assignmentCycle">Assigned Bidder Date</label>
+          <span>{assignedBidderDate}</span>
+        </div>
+      </div>
+    </>
+  );
+
+
+  const editView = () => (
```
Comment (M00staff) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:309]: I think for the Create New Assignment we only listed a few fields as required, and that was it as far as Form Validation went - should that be the same here?

There isn't any kind of validation at the moment

File: src/Components/CycleManagement/CycleSearchCard.jsx (hunk 9 above)
Comment (elizabeth-jimenez) [reply src/Components/CycleManagement/CycleSearchCard.jsx:7]: i think the original UI more matched the rest of the new pages (more complicated, but existing patterns)

They may not even notice that this design deviates tho, so we'll see 🤷‍♀️

**Comment by elizabeth-jimenez** [comment]
image
we'll want a Back button for when they don't Save and Return

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx (hunk 15 above)
Comment (elizabeth-jimenez) [reply src/Components/CycleManagement/AssignmentCycleEdit.jsx:366]: no validation if no one told u to add any - if anyone wants, they should provide exactly what they want validated

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx (hunk 4 above)
Comment (elizabeth-jimenez) [reply src/Components/CycleManagement/AssignmentCycleEdit.jsx:110]: liar

File: src/Constants/SystemMessages.js [hunk 16]
```diff
@@ -318,10 +318,15 @@ export const ASSIGNMENT_CYCLE_DELETE_SUCCESS = 'The Assignment Cycle has been de
...
-export const ASSIGNMENT_CYCLE_POST_ERROR = 'There was an issue attempting to create this Assignment Cycle. Please try again.';
+export const ASSIGNMENT_CYCLE_CREATE_SUCCESS_TITLE = 'Assignment Cycle Created';
+export const ASSIGNMENT_CYCLE_CREATE_SUCCESS = 'The Assignment Cycle has been created';
+export const ASSIGNMENT_CYCLE_CREATE_ERROR_TITLE = 'Assignment Cycle Create Error';
+export const ASSIGNMENT_CYCLE_CREATE_ERROR = 'There was an issue attempting to create this Assignment Cycle. Please try again.';
+
+export const ASSIGNMENT_CYCLE_POST_POSITIONS_TITLE = 'Open Positions Posted';
+export const ASSIGNMENT_CYCLE_POST_POSITIONS_SUCCESS = 'The Open Positions for this Assignment Cycle have been posted';
+export const ASSIGNMENT_CYCLE_POST_POSITIONS_ERROR_TITLE = 'Post Open Positions Error';
+export const ASSIGNMENT_CYCLE_POST_POSITIONS_ERROR = 'There was an issue attempting to post the Open Positions for this Assignment Cycle. Please try again.';
```
Comment (elizabeth-jimenez) [inline src/Constants/SystemMessages.js:329]: 🙌

File: src/sass/_positionSearch.scss [hunk 17]
```diff
@@ -256,6 +256,15 @@
   }
 }
 
+.ace-heading {
+  display: flex;
+  justify-content: space-between;
+  margin-top: 15px;
+}
+.ace-wrapper {
+  padding: 0 15px 0 15px;
```
Comment (elizabeth-jimenez) [inline src/sass/_positionSearch.scss:265]: you can do padding-inline: 15px here, if you wanted

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 18]
```diff
@@ -0,0 +1,695 @@
...
+  const [bidDue, setBidDue] = useState(assignmentCycle?.dates_mapping?.BIDDUE?.begin_date);
+  const [bidHtf, setBidHtf] = useState(assignmentCycle?.dates_mapping?.BIDHTF?.begin_date);
+  const [bidMds, setBidMds] = useState(assignmentCycle?.dates_mapping?.BIDMDS?.begin_date);
+  const [bidOrg, setBidOrg] = useState(assignmentCycle?.dates_mapping?.BIDORG?.begin_date);
+  const [bidStart, setBidStart] = useState(assignmentCycle?.dates_mapping?.BIDSTART?.begin_date);
+  const [burBid, setBurBid] = useState(assignmentCycle?.dates_mapping?.BURBID?.begin_date);
+  const [burEarly, setBurEarly] = useState(assignmentCycle?.dates_mapping?.BUREARLY?.begin_date);
+  const [burPos, setBurPos] = useState(assignmentCycle?.dates_mapping?.BURPOS?.begin_date);
+  const [burPrebd, setBurPrebd] = useState(assignmentCycle?.dates_mapping?.BURPREBD?.begin_date);
+  const [assignedBidderDate, setAssignedBidderDate] = useState(assignmentCycle?.dates_mapping?.PANLASG?.begin_date);
```
Comment (elizabeth-jimenez) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:63]: Let's match these states to how they are named in NewAssignmentCycle.jsx, basically what they are being mapped to in `cycleData` - it'll help with debugging in the future when we can easily connect the UI to the code

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 19]
```diff
@@ -0,0 +1,695 @@
...
+              overrideLifecycle
+            />
+          </span>
+        </div>
+        <div className="ac-sections">
+          <label htmlFor="assignmentCycle">Cycle Status</label>
+          <span>{cycleStatus?.label}</span>
+        </div>
+        <div className="ac-sections">
+          <label htmlFor="assignmentCycle">Cycle Boundry Dates</label>
```
Comment (elizabeth-jimenez) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:294]:
```suggestion
          <label htmlFor="assignmentCycle">Cycle Boundary Dates</label>
```

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 20]
```diff
@@ -0,0 +1,695 @@
...
+            <CheckBox
+              id="postViewable"
+              name="postViewable"
+              value={postViewable === 'Y'}
+              onChange={e => setPostViewable(e ? 'Y' : 'N')}
+            />
+          </span>
+        </div>
+        <div className="ac-sections">
+          <label htmlFor="cycleCategory">Cycle Category</label>
```
Comment (elizabeth-jimenez) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:404]: Cycle Category shifts places from the read view - between Assignment Cycle and Exclusive Position

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 21]
```diff
@@ -0,0 +1,695 @@
...
+          <label htmlFor="assignmentCycle">Cycle Status</label>
+          <span>{cycleStatus?.label}</span>
+        </div>
+        <div className="ac-sections">
+          <label htmlFor="assignmentCycle">Cycle Boundry Dates</label>
+          <span>{cycleDates?.[0]} - {cycleDates?.[1]}</span>
+        </div>
+        <div className="ac-sections">
+          <label htmlFor="assignmentCycle">6 Month Language Dates</label>
+          <span>{sixMonthLang?.[0]} - {sixMonthLang?.[0]}</span>
```
Comment (elizabeth-jimenez) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:299]:
```suggestion
          <span>{sixMonthLang?.[0]} - {sixMonthLang?.[1]}</span>
```

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 22]
```diff
@@ -0,0 +1,695 @@
...
+    }
+    return overlay;
+  };
+
+
+  return (
+    <div className="cycle-management-page position-search">
+      <div className="position-search--header">
+        <div className="expanded-content pt-20">
+          <ProfileSectionTitle title="Cycle Search" icon="cogs" className="xl-icon" />
```
Comment (elizabeth-jimenez) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:662]:
```suggestion
          <ProfileSectionTitle title="IDK what to change it to" icon="cogs" className="xl-icon" />
```

Comment (elizabeth-jimenez) [reply src/Components/CycleManagement/AssignmentCycleEdit.jsx:662]: oh i wonder if we call the previous page Cycle Search and then this page Cycle Management? it conflicts with our menu naming a bit, but this page deviates from other pages where we edit on the same Search page 🤔 idk

File: src/Components/CycleManagement/CycleSearchCard.jsx (hunk 13 above)
Comment (elizabeth-jimenez) [reply src/Components/CycleManagement/CycleSearchCard.jsx:32]: @M00staff any updates on this?

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx (hunk 6 above)
Comment (elizabeth-jimenez) [reply src/Components/CycleManagement/AssignmentCycleEdit.jsx:638]: any more insight on how we should proceed with this?

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 23]
```diff
@@ -0,0 +1,695 @@
...
+      setEditMode(false);
+      dispatch(cycleManagementUpdateCycleSuccess(false));
+      dispatch(cycleManagementAssignmentCycleFetchData(match?.params?.id));
+    }
+  }, [assignmentCycleUpdateSuccess]);
+
+  useEffect(() => {
+    if (assignmentCycleDeleteSuccess) {
+      dispatch(cycleManagementDeleteCycleSuccess(false));
+      history.push(`/profile/${isAO ? 'ao' : 'bureau'}/cyclemanagement`);
```
Comment (elizabeth-jimenez) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:83]: 🙌

File: src/Components/AoPage/AoPage.jsx [hunk 24]
```diff
@@ -23,6 +24,7 @@ const AoPage = () => (
       <Route path="/profile/ao/availablebidders" render={() => <AvailableBidderContainer isCDO={false} isAO />} />
       <Route path="/profile/ao/panelmeetingagendas/:pmID" render={() => <PanelMeetingAgendas isAO />} />
       <Route path="/profile/ao/cyclemanagement" render={() => <CycleManagement isAO />} />
+      <Route path="/profile/ao/assginmentcycle/:id" render={() => <AssignmentCycleEdit isAO />} />
```
Comment (elizabeth-jimenez) [inline src/Components/AoPage/AoPage.jsx:27]:
```suggestion
      <Route path="/profile/ao/assignmentcycle/:id" render={() => <AssignmentCycleEdit isAO />} />
```

Comment (elizabeth-jimenez) [reply src/Components/AoPage/AoPage.jsx:27]: typo

File: src/Components/BureauPage/BureauPage.jsx [hunk 25]
```diff
@@ -20,6 +21,7 @@ const BureauPage = () => {
       <Switch>
         <Route path="/profile/bureau/availablebidders" render={() => <AvailableBidderContainer isCDO={false} />} />
         <Route path="/profile/bureau/cyclemanagement" render={() => <CycleManagement isAO={false} />} />
+        <Route path="/profile/bureau/assginmentcycle/:id" render={() => <AssignmentCycleEdit isAO={false} />} />
```
Comment (elizabeth-jimenez) [inline src/Components/BureauPage/BureauPage.jsx:24]:
```suggestion
        <Route path="/profile/bureau/assignmentcycle/:id" render={() => <AssignmentCycleEdit isAO={false} />} />
```

File: src/Components/CycleManagement/CycleSearchCard.jsx [hunk 26]
```diff
@@ -28,186 +14,23 @@ const CycleSearchCard = (props) => {
     excl_position,
     post_view,
     isAO,
-    onEditModeSearch,
   } = props;
 
-  const cycleLink = `/profile/${isAO ? 'ao' : 'bureau'}/cyclepositionsearch/${id}`;
+  const cycleLink = `/profile/${isAO ? 'ao' : 'bureau'}/assginmentcycle/${id}`;
```
Comment (elizabeth-jimenez) [inline src/Components/CycleManagement/CycleSearchCard.jsx:19]:
```suggestion
  const cycleLink = `/profile/${isAO ? 'ao' : 'bureau'}/assignmentcycle/${id}`;
```

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx [hunk 27]
```diff
@@ -0,0 +1,695 @@
...
+        <div className="ac-sections">
+          <label htmlFor="cycleStatus">Cycle Status</label>
+          <span className="bs-validation-container">
+            <select
+              id="cycleStatus"
+              defaultValue={cycleStatus.value}
+              onChange={(e) => setCycleStatus(e.target.value)}
+              value={cycleStatus.value}
+            >
+              {cycleStatuses.length === 0 ?
```
Comment (elizabeth-jimenez) [inline src/Components/CycleManagement/AssignmentCycleEdit.jsx:430]: are we particularly at risk of cycleCategories and cycleStatuses not returning?

**Comment by elizabeth-jimenez** [comment]
IIRC the only required field was cycle dates, so we'll want to implement some kinda check there

**Comment by elizabeth-jimenez** [comment]
will want unique cancels for dispatches - currently some aren't canceling

File: src/actions/cycleManagement.js [hunk 28]
```diff
@@ -253,40 +221,126 @@ export function cycleManagementAssignmentCycleFetchDataSuccess(results) {
...
-export function cycleManagementAssignmentCycleFetchData() {
+export function cycleManagementAssignmentCycleFetchData(id) {
   return (dispatch) => {
     batch(() => {
       dispatch(cycleManagementAssignmentCycleFetchDataLoading(true));
       dispatch(cycleManagementAssignmentCycleFetchDataErrored(false));
     });
-    api().post('/placeholder/endpoint')
-      .then(() => {
+    api().post(`/fsbid/assignment_cycles/${id}/`)
```
Comment (elizabeth-jimenez) [inline src/actions/cycleManagement.js:230]:
```suggestion
    api().get(`/fsbid/assignment_cycles/${id}/`)
```

**Comment by elizabeth-jimenez** [comment]
When I first log in, the first time i click into an Assignment Cycle Details it loads and then stays at the Cycle Management page

**Comment by elizabeth-jimenez** [comment]
> When I first log in, the first time i click into an Assignment Cycle Details it loads and then stays at the Cycle Management page

Found it - you need to default ur state for cycleManagementDelete reducer to false

**Comment by elizabeth-jimenez** [comment]
Changes Requested
All the things pointed out above, unless they say something like 'optional' or 'if you want'

File: src/Components/CycleManagement/CycleSearchCard.jsx (hunk 13 above)
Comment (M00staff) [reply src/Components/CycleManagement/CycleSearchCard.jsx:32]: yep - only an Admin can `edit` but anyone can should be able to see the Cycle Details 

will update

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx (hunk 22 above)
Comment (M00staff) [reply src/Components/CycleManagement/AssignmentCycleEdit.jsx:662]: I'm cool with that!

File: src/Components/CycleManagement/CycleSearchCard.jsx (hunk 13 above)
Comment (M00staff) [reply src/Components/CycleManagement/CycleSearchCard.jsx:32]: updated

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx (hunk 15 above)
Comment (M00staff) [reply src/Components/CycleManagement/AssignmentCycleEdit.jsx:371]: added `disableSave` to make sure all the dates & name have values 🤷

better safe than sorry - should also keep from janking up the request

File: src/Components/AoPage/AoPage.jsx (hunk 24 above)
Comment (M00staff) [reply src/Components/AoPage/AoPage.jsx]: fixed!

File: src/Components/BureauPage/BureauPage.jsx (hunk 25 above)
Comment (M00staff) [reply src/Components/BureauPage/BureauPage.jsx]: fixed!

File: src/Components/CycleManagement/CycleSearchCard.jsx (hunk 26 above)
Comment (M00staff) [reply src/Components/CycleManagement/CycleSearchCard.jsx]: fixed!

File: src/Components/CycleManagement/AssignmentCycleEdit.jsx (hunk 6 above)
Comment (M00staff) [reply src/Components/CycleManagement/AssignmentCycleEdit.jsx:674]: image
🤷

**Comment by M00staff** [comment]
should be good for another look @elizabeth-jimenez

**Comment by elizabeth-jimenez** [comment]
😅 When we select filters that have no results, our ability to de-select them disappears
image

**Comment by elizabeth-jimenez** [comment]
Optional to fix: when we delete a cycle, we are sent back to the Cycle Search and if we had filtered for a position, then it will still show on the screen

**Comment by elizabeth-jimenez** [comment]
Approved Pending Feedback: 
- 😅 When we select filters that have no results, our ability to de-select them disappears

**Review by elizabeth-jimenez — APPROVED** [review]

**Comment by M00staff** [comment]
moved the overlay & set the create assignment to require all fields!

🙌

**Comment by elizabeth-jimenez** [comment]
> moved the overlay & set the create assignment to require all fields!
> 
> 🙌

Nice job! I didnt even expect the Create stuff in this PR/ticket 🦜
